src/sinks/logdna.rs

```diff
@@ -64,11 +64,13 @@ pub enum Encoding {
 impl SinkConfig for LogdnaConfig {
     fn build(&self, cx: SinkContext) -> crate::Result<(super::RouterSink, super::Healthcheck)> {
         let request_settings = self.request.unwrap_with(&TowerRequestConfig::default());
-        let batch_settings = self.batch.unwrap_or(bytesize::mib(10u64), 1);
+        let batch_settings = self
+            .batch
+            .parse_with_bytes(bytesize::mib(10u64), 100_000, 1)?;
 
         let sink = BatchedHttpSink::new(
             self.clone(),
-            JsonArrayBuffer::default(),
+            JsonArrayBuffer::new(batch_settings),
             request_settings,
             batch_settings,
```

Comment:
It's a little funky that we pass the `batch_settings` both to the batch itself and to the sink.

Reply:
I was unhappy with that too. I worked a bit on having `BatchSink::new` etc take a closure to which it would then pass the batch settings to create a new batch, but it was looking like it would add more confusion than it would solve. With the exception of `MetricBuffer` that could also be used to get rid of the `fresh` and `fresh_replace` methods, but I don't see a path to make that work. The only part of `BatchSettings` that the sink actually uses is the timeout, so I could split that part out in the calling convention.

Reply:
Yeah, that's interesting. From a user's perspective, size and time limits are both just batch settings, but internally they're handled by two different systems. It could make sense to split the structs accordingly.

Reply:
I went ahead and broke out the size settings to be able to pass them separately.